packages/common/src/projects/HubProjects.ts

```diff
@@ -0,0 +1,232 @@
+ * Create a new Hub Project item
+ *
+ * Minimal properties are name and org
+ *
+ * @param project
+ * @param requestOptions
+ */
+export async function createProject(
+  partialProject: Partial<IHubProject>,
+  requestOptions: IUserRequestOptions
+): Promise<IHubProject> {
+  // merge incoming with the default
+  // this expansion solves the typing somehow
+  const project = { ...DEFAULT_PROJECT, ...partialProject };
+
+  // Create a slug from the title if one is not passed in
+  if (!project.slug) {
+    project.slug = constructSlug(project.name, project.org.key);
+  }
+  // Ensure slug is  unique
+  project.slug = await getUniqueSlug(project.slug, requestOptions);
+  // add slug to keywords
+  project.typeKeywords = setSlugKeyword(project.typeKeywords, project.slug);
+  // Map project object onto a default project Model
+  const mapper = new PropertyMapper<Partial<IHubProject>>(
+    getProjectPropertyMap()
+  );
+  // create model from object, using the default model as a starting point
+  let model = mapper.objectToModel(project, cloneObject(DEFAULT_PROJECT_MODEL));
+  // create the item
+  model = await createModel(model, requestOptions);
+  // map the model back into a IHubProject
+  const newProject = mapper.modelToObject(model, {});
+  // and return it
+  return newProject as IHubProject;
+}
+
+/**
+ * Update a Hub Project
+ * @param project
+ * @param requestOptions
+ */
+export async function updateProject(
+  project: IHubProject,
+  requestOptions: IUserRequestOptions
+): Promise<IHubProject> {
+  // verify that the slug is unique
+  // TODO: if the slug has not changed, we don't want to do this because
+  // the existing item will cause this to increment
+  project.slug = await getUniqueSlug(project.slug, requestOptions);
+  // get the backing item & data
+  const model = await getModel(project.id, requestOptions);
+  // create the PropertyMapper
+  const mapper = new PropertyMapper<Partial<IHubProject>>(
+    getProjectPropertyMap()
+  );
+  // Note: Although we are fetching the model, and applying changes onto it,
+  // we are not attempting to handle "concurrent edit" conflict resolution
+  // but this is where we would apply that sort of logic
+  const modelToUpdate = mapper.objectToModel(project, model);
+  // update the backing item
+  const updatedModel = await updateModel(modelToUpdate, requestOptions);
+  // now map back into a project and return that
+  const updatedProject = mapper.modelToObject(updatedModel, project);
+  // the casting is needed because modelToObject returns a `Partial<T>`
+  // where as this function returns a `T`
+  return updatedProject as IHubProject;
+}
+
+/**
+ * Get a Hub Project by id or slug
+ * @param identifier item id or slug
+ * @param requestOptions
+ */
+export function getProject(
+  identifier: string,
+  requestOptions: IRequestOptions
+): Promise<IHubProject> {
+  let getPrms;
+  if (isGuid(identifier)) {
+    // get item by id
+    getPrms = getModel(identifier, requestOptions);
```

Comment:
This may be fine for now, but I suspect that projects, like content, will end up needing other enrichments, like `grouopIds`, `owner`, and `org`. 

For content I'm going to first fetch the item, then call `fetchContentEnrichments(item, fetcEnrichmentOptions)` where it takes an [IFetchEnrichmentOptions](https://github.com/Esri/hub.js/blob/20e4444ed460e7abd84883c7265f574919dd37fd/packages/content/src/enrichments.ts#L514-L522) and will return something like `{ data, groupIds, owner, org, metadata?, server?, layers? }` as appropriate based on the item type. I could have that in turn call `fetchItemEnrichments(item, options), where `options.enrichments` is the explicit list of enrichments to fetch. You could then use that here to get data and whatever else you end up needing.

Reply:
Agreed and I like that a lot.